### PR TITLE
feat(samples): Components samples

### PR DESCRIPTION
**Pull request by azabbasi**
Add Components samples 
Move serialization package from implementation to core. We have models here that we are exposing to the user so they should not be in the implementation package.

### CODE REVIEW

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/ComponentSyncSamples.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            .replace(SamplesConstants.ModelId, modelId)
+            .replace(SamplesConstants.ComponentId, componentModelId);
+
+        List<String> modelsList = new ArrayList<>(Arrays.asList(newComponentModelPayload, newModelPayload));
+
+        ConsoleLogger.printHeader("Create Models");
+        // We now create all the models (including components)
+        client.createModels(modelsList);
+
+        ConsoleLogger.print("Created models: " + componentModelId + " and " + modelId);
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/ComponentSyncSamples.java:72]: wouldn't we want to iterate over the created models list and use those Ids to log the success statement here?

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/helpers/ConsoleLogger.java [hunk 2]
```diff
@@ -7,21 +7,27 @@
...
     private static final String ANSI_CYAN = "\u001B[36m";
 
-    public static void PrintHeader(String message) {
+    public static void printHeader(String message) {
         System.out.println(ANSI_CYAN + "\n===" + message.toUpperCase() + "===\n" + ANSI_RESET);
     }
 
-    public static void PrintSuccess(String message) {
-        System.out.println(ANSI_GREEN + message + ANSI_RESET);
+    public static void print(String message) {
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/helpers/ConsoleLogger.java:17]: Could you add some comments to these helper methods, describing their user-cases. 
Also, just FYI - Jonathan recommended that we don't use println, I have asked for clarification on if we should use printf or logger.

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/serialization/ComponentMetadata.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An optional helper class for deserializing a digital twin
+ */
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/serialization/ComponentMetadata.java:11]: This class should be annotated with `@Fluent`.

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/serialization/ModelProperties.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Properties on a component that adhere to a specific model.
+ */
+public class ModelProperties {
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/serialization/ModelProperties.java:13]: `@Fluent` here, as well

**Review by abhipsaMisra — APPROVED** [review]

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClient.java [hunk 5]
```diff
@@ -221,6 +226,10 @@ public void deleteDigitalTwin(String digitalTwinId)
         return digitalTwinsAsyncClient.deleteDigitalTwinWithResponse(digitalTwinId, options, context).block();
     }
 
+    //==================================================================================================================================================
+    // Relationship APIs
```
Comment (timtay-microsoft) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClient.java:230]: While you are sprucing this up, we can actually do region sections in java, too.

```java
//region Models APIs
...
//endregion Models APIs
```

Intellij will let you collapse the code just like how C# does

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/serialization/ModelProperties.java (hunk 4 above)
Comment (timtay-microsoft) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/serialization/ModelProperties.java:13]: Does this add any functionality? I haven't used this annotation before

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClient.java (hunk 5 above)
Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClient.java:230]: much better.

**Review by timtay-microsoft — APPROVED** [review]

**Pull request merged**